framework/src/media/FileOutputDataSource.cpp

```diff
@@ -88,17 +94,18 @@ bool FileOutputDataSource::open()
 bool FileOutputDataSource::close()
 {
 	switch (getAudioType()) {
-		case AUDIO_TYPE_WAVE:
-			unsigned int fileSize;
-			fileSize = ftell(mFp);
-			if (fileSize < 0) {
+		case AUDIO_TYPE_WAVE: {
+			long ret = ftell(mFp);
```

Comment:
As I know, bofore call fsync(when sector is filled with data), ftell will return wrong value.
How about maintain variable and update it when write called?

Reply:
`USER -> fwrite -> STDIO-BUFFER -> fflush -> SYSTEM-BUFFER -> fsync -> FLASH`
There may be data cached in STDIO-BUFFER has not be flushed into SYSTEM-BUFFER,
and `ftell` get position(file size) from SYSTEM-BUFFER, so the size may be less than USER write.
Is it what you mean?

Reply:
@Taejun-Kwon  what's your opinion?

Reply:
fflush write remained data in buffer, but sync is called if next sector is needed because of data is bigger than remained area in current sector.
But As per you wrote in above, ftell using position of stream buffer, I think it is enough.